components/ContractViewer/AstProcessor.ts

```diff
@@ -0,0 +1,92 @@
+import solParser from '@solidity-parser/parser'
+import * as AstTypes from '@solidity-parser/parser/src/ast-types'
+
+import { ParseResult } from './DeploymentInfo'
+
+export type ContractArtifact<T = AstTypes.BaseASTNode> = {
+  id: number
+  node: T
+  // undefined for global scope
+  scope?: AstTypes.ContractDefinition | AstTypes.FunctionDefinition
+}
+
+// this represents a single contract/interface/library definition block
+export class SourceDefinition {
+  // name: string = ''
+  // filepath: string = ''
+  // c3 inheritance / ast node ref?

```

Comment:
Can you clarify or remove the comments here ?

Reply:
if you have a better location for declaration of intents I am open to hear.
otherwise over time these plans will be forgotten.

Reply:
I just meant that I didn't fully understand the comment so was wondering if you could add some more context.

Is the comment meant to say that you want to keep track of the inheritance of the contract with the name and filepath ? 
(if so could you expand upon why that is useful - just so others can follow the thought process)

Reply:
ok I will remove the comments.